editor/components/block-toolbar/index.js

```diff
@@ -16,22 +16,16 @@ import BlockSwitcher from '../block-switcher';
 import { getBlockMode, getSelectedBlock } from '../../store/selectors';
 
 function BlockToolbar( { block, mode } ) {
-	if ( ! block || ! block.isValid ) {
+	if ( ! block || ! block.isValid || mode !== 'visual' ) {
```

Comment:
Granted the issue was here before but I think the `mode` should not be checked here but more in the `edit-post` folder where the `BlockToolbar` is inserted (as it's layout specific)

Reply:
>Granted the issue was here before but I think the `mode` should not be checked here but more in the `edit-post` folder where the `BlockToolbar` is inserted (as it's layout specific)

Are you referring to the "Visual vs. Text" distinction on the post editor layout, or (as implemented here) the "Edit as Visual vs Edit as HTML" distinction on a block?

Reply:
Oh sorry! I misread the PR. You're totally right and the block mode makes sense heree.

Reply:
>Oh sorry! I misread the PR. You're totally right and the block mode makes sense here.

I mean, it's good to highlight the confusion at least caused by overloading the use of the word Visual 😆